src/Microsoft.Health.Client/OAuth2ClientCertificateCredentialProvider.cs

```diff
@@ -23,36 +23,43 @@ public class OAuth2ClientCertificateCredentialProvider : CredentialProvider
         private const string JwtAssertionType = "urn:ietf:params:oauth:client-assertion-type:jwt-bearer";
         private const string Rs256Algorithm = "RS256";
 
-        private readonly OAuth2ClientCertificateCredentialConfiguration _oAuth2ClientCertificateCredentialConfiguration;
+        private readonly IOptionsMonitor<OAuth2ClientCertificateCredentialConfiguration> _oAuth2ClientCertificateCredentialConfigurationMonitor;
         private readonly HttpClient _httpClient;
+        private readonly string _optionsName;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="OAuth2ClientCertificateCredentialProvider"/> class.
         /// This class is used to obtain a token for the configured resource via the OAuth2 token endpoint via client certificate credentials.
         /// </summary>
-        /// <param name="oAuth2ClientCertificateCredentialConfiguration">The configuration to use when obtaining a token.</param>
+        /// <param name="oAuth2ClientCertificateCredentialConfigurationMonitorMonitor">The configuration to use when obtaining a token.</param>
         /// <param name="httpClient">The <see cref="HttpClient" /> to use when calling the token uri.</param>
-        public OAuth2ClientCertificateCredentialProvider(IOptions<OAuth2ClientCertificateCredentialConfiguration> oAuth2ClientCertificateCredentialConfiguration, HttpClient httpClient)
+        /// <param name="optionsName">Optional name to use when retrieving options from the IOptionsMonitor</param>
+        public OAuth2ClientCertificateCredentialProvider(IOptionsMonitor<OAuth2ClientCertificateCredentialConfiguration> oAuth2ClientCertificateCredentialConfigurationMonitorMonitor, HttpClient httpClient, string optionsName = null)
         {
-            EnsureArg.IsNotNull(oAuth2ClientCertificateCredentialConfiguration?.Value, nameof(oAuth2ClientCertificateCredentialConfiguration));
+            EnsureArg.IsNotNull(oAuth2ClientCertificateCredentialConfigurationMonitorMonitor, nameof(oAuth2ClientCertificateCredentialConfigurationMonitorMonitor));
             EnsureArg.IsNotNull(httpClient, nameof(httpClient));
 
             _httpClient = httpClient;
-            _oAuth2ClientCertificateCredentialConfiguration = oAuth2ClientCertificateCredentialConfiguration.Value;
+            _oAuth2ClientCertificateCredentialConfigurationMonitor = oAuth2ClientCertificateCredentialConfigurationMonitorMonitor;
+            _optionsName = optionsName;
```

Comment:
Same comment about normalizing this value to `null`, unless we want to allow options called `""`